src/dotty/tools/dotc/typer/EtaExpansion.scala

```diff
@@ -48,13 +48,11 @@ object EtaExpansion {
   }
 
   /** Lift a function argument, stripping any NamedArg wrapper */
```

Comment:
We do strip it, but then rewrap it around something else. I meant to do the change, but then decided it's too much busywork for too minor an improvement.